u_conn.go

```diff
@@ -749,9 +749,10 @@ func (c *Conn) utlsConnectionStateLocked(state *ConnectionState) {
 
 type utlsConnExtraFields struct {
 	// Application Settings (ALPS)
-	hasApplicationSettings   bool
-	peerApplicationSettings  []byte
-	localApplicationSettings []byte
+	hasApplicationSettings       bool
+	peerApplicationSettings      []byte
+	localApplicationSettings     []byte
+	applicationSettingsCodepoint uint16
```

Comment:
Any chance we can use `applicationSettingsCodepoint != 0` as `hasApplicationSettings` and remove it?